src/transformers/tokenization_marian.py

```diff
@@ -0,0 +1,162 @@
+import warnings
+from typing import List, Optional
+
+import sentencepiece
+
+from .file_utils import S3_BUCKET_PREFIX, load_json
+from .tokenization_utils import BatchEncoding, PreTrainedTokenizer
+
+
+vocab_files_names = {
+    "source_spm": "source.spm",
+    "target_spm": "target.spm",
+    "vocab": "vocab.json",
+    "tokenizer_config_file": "tokenizer_config.json",
+}
+MODEL_NAMES = ("marian-en-de",)
+PRETRAINED_VOCAB_FILES_MAP = {
+    k: {m: f"{S3_BUCKET_PREFIX}/opus/{m}/{fname}" for m in MODEL_NAMES} for k, fname in vocab_files_names.items()
+}
+# Example URL https://s3.amazonaws.com/models.huggingface.co/bert/opus/marian-en-de/vocab.json
+
+
+class MarianSentencePieceTokenizer(PreTrainedTokenizer):
```

Comment:
Why `MarianSentencePieceTokenizer` instead of `MarianTokenizer`? Are we expecting other Marian tokenizers? (We don't have `AlbertSentencePieceTokenizer` or `XLNetSentencePieceTokenizer` even though they both are SP tokenizers)

Reply:
Good Q. About 10% of the Opus models don't use sentencepiece, they use BPE. Was going to tackle those in a separate PR.